backend/internal/system/crypto/hash/hashprovider.go

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Package hash provides generic hashing utilities for sensitive data.
+package hash
+
+import (
+	"crypto/pbkdf2"
+	"crypto/sha256"
+	"encoding/hex"
+
+	"github.com/asgardeo/thunder/internal/system/config"
+	"github.com/asgardeo/thunder/internal/system/log"
+)
+
+// HashProviderInterface defines the interface for hashing and verifying data.
+type HashProviderInterface interface {
```

Comment:
Even though we say this is a provider, it's responsibility is for a service. So we can have something like this.

```go
type HashProviderInterface interface {
    GetHashService() HashServiceInterface
}

type HashProvider struct {}
// Implement HashProvider

type HashService interface{
	Hash(data []byte, salt []byte) string
	Verify(data []byte, salt []byte, expectedHash string) bool
}

type SHA256HashService struct{}
// Implement SHA256HashService

type PBKDF2HashService struct {
    Iterations int
    KeyLen     int
}
// Implement PBKDF2HashService
```

But the term provider may more suitable for hashing rather than saying service.
@darshanasbg wdyt?

Reply:
@hwupathum talked to me before raising this PR. What we discussed was:
> Hashing functionality we going to onboard, does not involve with any state (that usually maintained by a service layer).. 
> This is purely a utility. ie Give the input in one form, and it convert to some other

Considering this, it was considered to keep this as a hashing provider.. rather a hashing service.

Rather this provides